Userland/DevTools/UserspaceEmulator/Emulator_syscalls.cpp

```diff
@@ -689,6 +710,20 @@ int Emulator::virt$getsockopt(FlatPtr params_addr)
     TODO();
 }
 
+int Emulator::virt$getsockname(FlatPtr params_addr)
+{
+    Syscall::SC_getsockname_params params;
+    mmu().copy_from_vm(&params, params_addr, sizeof(params));
+    return getsockname(params.sockfd, params.addr, params.addrlen);
+}
+
+int Emulator::virt$getpeername(FlatPtr params_addr)
+{
+    Syscall::SC_getpeername_params params;
+    mmu().copy_from_vm(&params, params_addr, sizeof(params));
+    return getpeername(params.sockfd, params.addr, params.addrlen);
+}
```

Comment:
We're not copying the names back into VM memory here.

Reply:
Agreed. That would help.

Reply:
Fixed.